pkgs/applications/misc/blender/default.nix

```
@@ -8,144 +8,168 @@
-      "-DWITH_SDL=OFF"
-      "-DWITH_OPENCOLORIO=ON"
-      "-DWITH_OPENSUBDIV=ON"
-      "-DPYTHON_LIBRARY=${python.libPrefix}"
-      "-DPYTHON_LIBPATH=${python}/lib"
-      "-DPYTHON_INCLUDE_DIR=${python}/include/${python.libPrefix}"
-      "-DPYTHON_VERSION=${python.pythonVersion}"
-      "-DWITH_PYTHON_INSTALL=OFF"
-      "-DWITH_PYTHON_INSTALL_NUMPY=OFF"
-      "-DPYTHON_NUMPY_PATH=${python3Packages.numpy}/${python.sitePackages}"
-      "-DWITH_OPENVDB=ON"
-      "-DWITH_TBB=ON"
-      "-DWITH_IMAGE_OPENJPEG=ON"
-      "-DWITH_OPENCOLLADA=${if colladaSupport then "ON" else "OFF"}"
-    ]
-    ++ optionals stdenv.isDarwin [
-      "-DWITH_CYCLES_OSL=OFF" # requires LLVM
-      "-DWITH_OPENVDB=OFF" # OpenVDB currently doesn't build on darwin
-
-      "-DLIBDIR=/does-not-exist"
-    ]
-    # Clang doesn't support "-export-dynamic"
-    ++ optional stdenv.cc.isClang "-DPYTHON_LINKFLAGS="
-    ++ optional jackaudioSupport "-DWITH_JACK=ON"
-    ++ optional cudaSupport "-DWITH_CYCLES_CUDA_BINARIES=ON";
-
-  NIX_CFLAGS_COMPILE = "-I${ilmbase.dev}/include/OpenEXR -I${python}/include/${python.libPrefix}";
-
-  # Since some dependencies are built with gcc 6, we need gcc 6's
-  # libstdc++ in our RPATH. Sigh.
-  NIX_LDFLAGS = optionalString cudaSupport "-rpath ${stdenv.cc.cc.lib}/lib";
-
-  enableParallelBuilding = true;
-
-  blenderExecutable =
-    placeholder "out" + (if stdenv.isDarwin then "/Blender.app/Contents/MacOS/Blender" else "/bin/blender");
-  # --python-expr is used to workaround https://developer.blender.org/T74304
-  postInstall = ''
-    wrapProgram $blenderExecutable \
-      --prefix PYTHONPATH : ${python3Packages.numpy}/${python.sitePackages} \
-      --add-flags '--python-use-system-env'
-  '';
+    meta = with stdenv.lib; {
+      description = "3D Creation/Animation/Publishing System";
+      homepage = "https://www.blender.org";
+      # They comment two licenses: GPLv2 and Blender License, but they
+      # say: "We've decided to cancel the BL offering for an indefinite period."
+      license = licenses.gpl2Plus;
+      platforms = [ "x86_64-linux" "x86_64-darwin" ];
+      maintainers = with maintainers; [ goibhniu veprbl ];
+    };
+  };
+
+  wrapped = let
+    blenderExecutable = (if stdenv.isDarwin then "Blender.app/Contents/MacOS/Blender" else "bin/blender");
+  in runCommand unwrapped.name {
+    nativeBuildInputs = [
+      makeWrapper
+    ];
+    passthru = {
+      inherit unwrapped;
+      inherit (unwrapped) meta;
+    };
 
-  # Set RUNPATH so that libcuda and libnvrtc in /run/opengl-driver(-32)/lib can be
-  # found. See the explanation in libglvnd.
-  postFixup = optionalString cudaSupport ''
-    for program in $out/bin/blender $out/bin/.blender-wrapped; do
-      isELF "$program" || continue
-      addOpenGLRunpath "$program"
-    done
+  } # --python-expr is used to workaround https://developer.blender.org/T74304
+  ''
+    makeWrapper "${unwrapped}/${blenderExecutable}" "$out/${blenderExecutable}" \
```

```suggestion
    makeWrapper "$unwrapped/${blenderExecutable}" "$out/${blenderExecutable}" \
```
This should allow to override the unwrapped blender.